pytorch_toolkit/formula_recognition/README.md

```diff
@@ -0,0 +1,188 @@
+# PyTorch realization of the Formula Recognition
+
+This code is based on a [PyTorch realization](https://github.com/luopeixiang/im2latex/) of the code from the original [repository](https://github.com/harvardnlp/im2markup/).
+Models code is designed to enable ONNX\* export and inference on CPU\GPU via OpenVINO™.
+
+## Setup
+
+### Prerequisites
+
+* Ubuntu\* 16.04
+* Python\* 3.7 or newer
+* PyTorch\* (1.4.0)
+* OpenVINO™ 2020.4 with Python API
+
+### Optional prerequisites
+
+#### Install required packages for evaluation
+These packages are used for rendering images while evaluation and demo.
+
+```bash
+bash sudo apt-get update &&
+  apt-get install -y --no-install-recommends \
+    texlive \
+    imagemagick \
+    ghostscript
+```
+
+### Installation
+
+Create and activate virtual environment:
+
+```bash
+bash init_venv.sh
+```
+
+### Download Datasets
+
+Dataset format is similar to [im2latex-100k](https://zenodo.org/record/56198#.X2NDQ2gzaUl). Main structure of the dataset is following:
+* `formulas_file` - file with one formula per line
+* `images_folder` - folder containing input images
+* `split_file` - this file contains `image_name` (tab symbol) `formula_idx` per line connecting corresponding index of the formula in the file with formulas and particular image with `image_name`. Example:
+    ```
+    11.png  11
+    34.png  34
+    ...
+    ```
+    There should be at least two such files: `train_filter.lst` and `validate_filter.lst`
+
+> **NOTE**:
+> By default the following structure of the dataset is assumed:
+> `images_processed` - folder with images
+> `formulas.norm.lst` - file with preprocessed formulas, for details, refer to [im2markup](https://github.com/harvardnlp/im2markup) repository
+> `validate_filter.lst` and `train_filter.lst` - corresponding splits of the data.
+
+
+## Training
+
+To train formula recognition model run:
+
+```bash
+python tools/train.py --config configs/medium_config.yml --work_dir <path to work dir>
+```
+Work dir is used to store information about learning: saved model checkpoints, logs.
+
+### Description of possible options in config:
+The config file is divided into 5 sections: common, train, eval, export, demo. Common parameters (like path to the model) are stored, respectively, in common section. Unique parameters (like learning rate) are stored in other specific sections.
+#### Common parameters:
+ - `backbone_config`:
+    * `arch`: type of the architecture (if backbone_type is resnet). For more details, please, refer to [ResnetLikeBackBone](im2latex/models/backbones/resnet.py)
+    * `disable_layer_3` and `disable_layer_4` - disables layer 3 and 4 in resnet-like backbone
```

Comment:
Could you add a short description what is it?

At least, I'd propose to add info something like that     
`(note that ResNet backbone from torchvision consists of 4 block of layers, each of them increase the number of channels and decrease the spatial dimensionality, these parameters allow to swithch off the last of them)`

Reply:
Added information about layer_3 & layer_4